workflow/controller/operator.go

```diff
@@ -2630,6 +2650,38 @@ func (woc *wfOperationCtx) executeResource(ctx context.Context, nodeName string,
 	return node, err
 }
 
+func (woc *wfOperationCtx) executeData(ctx context.Context, nodeName string, templateScope string, tmpl *wfv1.Template, orgTmpl wfv1.TemplateReferenceHolder, opts *executeTemplateOpts) (*wfv1.NodeStatus, error) {
```

Comment:
own file please

Reply:
Like wise to: https://github.com/argoproj/argo-workflows/pull/4958#discussion_r581580810. Perhaps we should have a large tech-debt PR to break up these large files? `workflow_types.go` could also benefit from this

Reply:
I don't feel strongly, but it woc is MASSIVE and you might prefer to find you code in a single focussed file